pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -103,6 +103,16 @@ func (c *Reconciler) reconcileDeployment(ctx context.Context, rev *v1alpha1.Revi
 					break
 				}
 			}
+			if cond := rev.Status.GetCondition(v1alpha1.RevisionConditionResourcesAvailable); cond != nil && cond.Status == corev1.ConditionFalse {
```

Comment:
So here you already have `ResourcesAvailable`  set to `false`, which you replace with another equal value, just with different message.
This seems strange (since the bug described that we're getting `Unknown`.

Reply:
@vagababov I'm actually getting [this behavior](https://github.com/knative/serving/issues/5103#issuecomment-523730216) when trying this out locally. That's right. Here I'm replacing the message with a more specific reason if the secret specified cannot be found. Is that what we want?

Reply:
No that's fine that you're setting this error (though we might be masking another error), but as I said, I think the bug mentions getting Unknown state, rather than False in the ready state. If it was already _false_ then it should have marked ready as `False`, but it didn't. Which means Ready is unaffected by this change.

Reply:
@vagababov it did mark ready as `False` when I tried this locally

Reply:
@vagababov At https://github.com/knative/serving/commit/f1137d13961e79ec9721fea4b2b63eed86e72b86, this is the behavior I'm getting
```
Spec:
  Container Concurrency:  0
  Containers:
    Env:
      Name:   TARGET
      Value:  Go Sample v1
    Image:    docker.io/taragu/helloworld-go
    Name:     user-container
    Readiness Probe:
      Success Threshold:  1
      Tcp Socket:
        Port:  0
    Resources:
    Volume Mounts:
      Mount Path:   /etc/db-user
      Name:         db-user-pass
      Read Only:    true
  Timeout Seconds:  300
  Volumes:
    Name:  db-user-pass
    Secret:
      Secret Name:  db-user-pass-invalid
Status:
  Conditions:
    Last Transition Time:  2019-10-09T15:45:43Z
    Message:               Requests to the target are being buffered as resources are provisioned.
    Reason:                Queued
    Severity:              Info
    Status:                Unknown
    Type:                  Active
    Last Transition Time:  2019-10-09T15:45:42Z
    Reason:                Deploying
    Status:                Unknown
    Type:                  ContainerHealthy
    Last Transition Time:  2019-10-09T15:47:43Z
    Reason:                PodInitializing
    Status:                False
    Type:                  Ready
    Last Transition Time:  2019-10-09T15:47:43Z
    Reason:                PodInitializing
    Status:                False
    Type:                  ResourcesAvailable
```
The `ResourcesAvailable` was Unknown for a while, before it times out and becomes false. I'm adding an additional check before getting the secrets that the reason should be `PodInitializing`. I think the reason `PodInitializing` is generic enough that we can override it with a more specific message only if we have trouble getting secrets. What do you think?